src/lib/vrf_evaluator/vrf_evaluator.ml

```diff
@@ -0,0 +1,408 @@
+open Core
+open Async
+open Signature_lib
+open Pipe_lib
+module Epoch = Mina_numbers.Length
+module Global_slot = Mina_numbers.Global_slot
+
+(*Slot number within an epoch*)
+module Slot = Mina_numbers.Global_slot
+
+(* Can extract both slot numbers and epoch number*)
+module Consensus_time = Consensus.Data.Consensus_time
+
+module State = struct
+  [%%versioned
+  module Stable = struct
+    [@@@no_toplevel_latest_type]
+
+    module V1 = struct
+      type t =
+        { block_producer_keys:
+            (Keypair.Stable.V1.t * Public_key.Compressed.Stable.V1.t) list }
```

Comment:
If we want to support adding keys without re-computing for the other keys, should we store a `Evaluator_status` against each keypair?